google/cloud/firestore/field_path.cc

```diff
@@ -63,10 +61,33 @@ FieldPath FieldPath::Append(FieldPath const& field_path) const {
 }
 
 std::string FieldPath::ToApiRepr() const {
+#if defined(__GNUC__) && __GNUC__ == 4 && __GNUC_MINOR__ == 8
```

Comment:
Do we like the "defined(\_\_GNUC\_\_)"?  If it wasn't defined, "\_\_GNUC\_\_ == 4" would be false anyway.  Or would that make this expression gratuitously different from other uses where that doesn't apply?

Reply:
There are a few other uses of `defined(__GNUC__)` in this file, so I would prefer to keep as-is for now. Later we can change all of them, I have no opinion on which style is "better".

Having said that, if we find ourselves testing for compiler names and versions in a lot of places we are doing something wrong.